{"matches":[{"vulnerability":{"id":"GHSA-5rjg-fvgr-3xxf","dataSource":"https://github.com/advisories/GHSA-5rjg-fvgr-3xxf","namespace":"github:language:python","severity":"High","urls":[],"description":"setuptools has a path traversal vulnerability in PackageIndex.download that leads to Arbitrary File Write","cvss":[],"epss":[{"cve":"CVE-2025-47273","epss":0.00435,"percentile":0.62062,"date":"2025-06-03"}],"fix":{"versions":["78.1.1"],"state":"fixed"},"advisories":[],"risk":0.32625},"relatedVulnerabilities":[{"id":"CVE-2025-47273","dataSource":"https://nvd.nist.gov/vuln/detail/CVE-2025-47273","namespace":"nvd:cpe","severity":"High","urls":["https://github.com/pypa/setuptools/blob/6ead555c5fb29bc57fe6105b1bffc163f56fd558/setuptools/package_index.py#L810C1-L825C88","https://github.com/pypa/setuptools/commit/250a6d17978f9f6ac3ac887091f2d32886fbbb0b","https://github.com/pypa/setuptools/issues/4946","https://github.com/pypa/setuptools/security/advisories/GHSA-5rjg-fvgr-3xxf","https://lists.debian.org/debian-lts-announce/2025/05/msg00035.html","https://github.com/pypa/setuptools/issues/4946"],"description":"setuptools is a package that allows users to download, build, install, upgrade, and uninstall Python packages. A path traversal vulnerability in `PackageIndex` is present in setuptools prior to version 78.1.1. An attacker would be allowed to write files to arbitrary locations on the filesystem with the permissions of the process running the Python code, which could escalate to remote code execution depending on the context. Version 78.1.1 fixes the issue.","cvss":[{"source":"security-advisories@github.com","type":"Secondary","version":"4.0","vector":"CVSS:4.0/AV:N/AC:L/AT:N/PR:N/UI:N/VC:N/VI:H/VA:N/SC:N/SI:N/SA:N/E:P/CR:X/IR:X/AR:X/MAV:X/MAC:X/MAT:X/MPR:X/MUI:X/MVC:X/MVI:X/MVA:X/MSC:X/MSI:X/MSA:X/S:X/AU:X/R:X/V:X/RE:X/U:X","metrics":{"baseScore":7.7},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2025-47273","epss":0.00435,"percentile":0.62062,"date":"2025-06-03"}]}],"matchDetails":[{"type":"exact-direct-match","matcher":"python-matcher","searchedBy":{"language":"python","namespace":"github:language:python","package":{"name":"setuptools","version":"44.0.0"}},"found":{"versionConstraint":"<78.1.1 (python)","vulnerabilityID":"GHSA-5rjg-fvgr-3xxf"},"fix":{"suggestedVersion":"78.1.1"}}],"artifact":{"id":"e32f7cbfb9974e4a","name":"setuptools","version":"44.0.0","type":"python","locations":[{"path":"/lib/python3.8/site-packages/setuptools-44.0.0.dist-info/METADATA","accessPath":"/lib/python3.8/site-packages/setuptools-44.0.0.dist-info/METADATA","annotations":{"evidence":"primary"}},{"path":"/lib/python3.8/site-packages/setuptools-44.0.0.dist-info/RECORD","accessPath":"/lib/python3.8/site-packages/setuptools-44.0.0.dist-info/RECORD","annotations":{"evidence":"supporting"}},{"path":"/lib/python3.8/site-packages/setuptools-44.0.0.dist-info/top_level.txt","accessPath":"/lib/python3.8/site-packages/setuptools-44.0.0.dist-info/top_level.txt","annotations":{"evidence":"supporting"}}],"language":"python","licenses":["UNKNOWN"],"cpes":["cpe:2.3:a:python:setuptools:44.0.0:*:*:*:*:*:*:*"],"purl":"pkg:pypi/setuptools@44.0.0","upstreams":[]}},{"vulnerability":{"id":"GHSA-r9hx-vwmv-q579","dataSource":"https://github.com/advisories/GHSA-r9hx-vwmv-q579","namespace":"github:language:python","severity":"High","urls":[],"description":"pypa/setuptools vulnerable to Regular Expression Denial of Service (ReDoS)","cvss":[{"type":"Secondary","version":"3.1","vector":"CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:N/I:N/A:H","metrics":{"baseScore":7.5,"exploitabilityScore":3.9,"impactScore":3.6},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2022-40897","epss":0.00318,"percentile":0.54328,"date":"2025-06-03"}],"fix":{"versions":["65.5.1"],"state":"fixed"},"advisories":[],"risk":0.2385},"relatedVulnerabilities":[{"id":"CVE-2022-40897","dataSource":"https://nvd.nist.gov/vuln/detail/CVE-2022-40897","namespace":"nvd:cpe","severity":"Medium","urls":["https://github.com/pypa/setuptools/blob/fe8a98e696241487ba6ac9f91faa38ade939ec5d/setuptools/package_index.py#L200","https://github.com/pypa/setuptools/commit/43a9c9bfa6aa626ec2a22540bea28d2ca77964be","https://github.com/pypa/setuptools/compare/v65.5.0...v65.5.1","https://lists.fedoraproject.org/archives/list/package-announce%40lists.fedoraproject.org/message/ADES3NLOE5QJKBLGNZNI2RGVOSQXA37R/","https://lists.fedoraproject.org/archives/list/package-announce%40lists.fedoraproject.org/message/YNA2BAH2ACBZ4TVJZKFLCR7L23BG5C3H/","https://pyup.io/posts/pyup-discovers-redos-vulnerabilities-in-top-python-packages/","https://pyup.io/vulnerabilities/CVE-2022-40897/52495/","https://security.netapp.com/advisory/ntap-20230214-0001/","https://security.netapp.com/advisory/ntap-20240621-0006/","https://github.com/pypa/setuptools/blob/fe8a98e696241487ba6ac9f91faa38ade939ec5d/setuptools/package_index.py#L200","https://github.com/pypa/setuptools/commit/43a9c9bfa6aa626ec2a22540bea28d2ca77964be","https://github.com/pypa/setuptools/compare/v65.5.0...v65.5.1","https://lists.fedoraproject.org/archives/list/package-announce%40lists.fedoraproject.org/message/ADES3NLOE5QJKBLGNZNI2RGVOSQXA37R/","https://lists.fedoraproject.org/archives/list/package-announce%40lists.fedoraproject.org/message/YNA2BAH2ACBZ4TVJZKFLCR7L23BG5C3H/","https://pyup.io/posts/pyup-discovers-redos-vulnerabilities-in-top-python-packages/","https://pyup.io/vulnerabilities/CVE-2022-40897/52495/","https://security.netapp.com/advisory/ntap-20230214-0001/","https://security.netapp.com/advisory/ntap-20240621-0006/"],"description":"Python Packaging Authority (PyPA) setuptools before 65.5.1 allows remote attackers to cause a denial of service via HTML in a crafted package or custom PackageIndex page. There is a Regular Expression Denial of Service (ReDoS) in package_index.py.","cvss":[{"source":"nvd@nist.gov","type":"Primary","version":"3.1","vector":"CVSS:3.1/AV:N/AC:H/PR:N/UI:N/S:U/C:N/I:N/A:H","metrics":{"baseScore":5.9,"exploitabilityScore":2.3,"impactScore":3.6},"vendorMetadata":{}},{"source":"134c704f-9b21-4f2e-91b3-4a467353bcc0","type":"Secondary","version":"3.1","vector":"CVSS:3.1/AV:N/AC:H/PR:N/UI:N/S:U/C:N/I:N/A:H","metrics":{"baseScore":5.9,"exploitabilityScore":2.3,"impactScore":3.6},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2022-40897","epss":0.00318,"percentile":0.54328,"date":"2025-06-03"}]}],"matchDetails":[{"type":"exact-direct-match","matcher":"python-matcher","searchedBy":{"language":"python","namespace":"github:language:python","package":{"name":"setuptools","version":"44.0.0"}},"found":{"versionConstraint":"<65.5.1 (python)","vulnerabilityID":"GHSA-r9hx-vwmv-q579"},"fix":{"suggestedVersion":"65.5.1"}}],"artifact":{"id":"e32f7cbfb9974e4a","name":"setuptools","version":"44.0.0","type":"python","locations":[{"path":"/lib/python3.8/site-packages/setuptools-44.0.0.dist-info/METADATA","accessPath":"/lib/python3.8/site-packages/setuptools-44.0.0.dist-info/METADATA","annotations":{"evidence":"primary"}},{"path":"/lib/python3.8/site-packages/setuptools-44.0.0.dist-info/RECORD","accessPath":"/lib/python3.8/site-packages/setuptools-44.0.0.dist-info/RECORD","annotations":{"evidence":"supporting"}},{"path":"/lib/python3.8/site-packages/setuptools-44.0.0.dist-info/top_level.txt","accessPath":"/lib/python3.8/site-packages/setuptools-44.0.0.dist-info/top_level.txt","annotations":{"evidence":"supporting"}}],"language":"python","licenses":["UNKNOWN"],"cpes":["cpe:2.3:a:python:setuptools:44.0.0:*:*:*:*:*:*:*"],"purl":"pkg:pypi/setuptools@44.0.0","upstreams":[]}},{"vulnerability":{"id":"GHSA-cx63-2mw6-8hw5","dataSource":"https://github.com/advisories/GHSA-cx63-2mw6-8hw5","namespace":"github:language:python","severity":"High","urls":[],"description":"setuptools vulnerable to Command Injection via package URL","cvss":[{"type":"Secondary","version":"3.1","vector":"CVSS:3.1/AV:N/AC:L/PR:N/UI:R/S:U/C:H/I:H/A:H","metrics":{"baseScore":8.8,"exploitabilityScore":2.9,"impactScore":5.9},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2024-6345","epss":0.00227,"percentile":0.45699,"date":"2025-06-03"}],"fix":{"versions":["70.0.0"],"state":"fixed"},"advisories":[],"risk":0.185005},"relatedVulnerabilities":[{"id":"CVE-2024-6345","dataSource":"https://nvd.nist.gov/vuln/detail/CVE-2024-6345","namespace":"nvd:cpe","severity":"High","urls":["https://github.com/pypa/setuptools/commit/88807c7062788254f654ea8c03427adc859321f0","https://huntr.com/bounties/d6362117-ad57-4e83-951f-b8141c6e7ca5","https://github.com/pypa/setuptools/commit/88807c7062788254f654ea8c03427adc859321f0","https://huntr.com/bounties/d6362117-ad57-4e83-951f-b8141c6e7ca5"],"description":"A vulnerability in the package_index module of pypa/setuptools versions up to 69.1.1 allows for remote code execution via its download functions. These functions, which are used to download packages from URLs provided by users or retrieved from package index servers, are susceptible to code injection. If these functions are exposed to user-controlled inputs, such as package URLs, they can execute arbitrary commands on the system. The issue is fixed in version 70.0.","cvss":[{"source":"security@huntr.dev","type":"Secondary","version":"3.0","vector":"CVSS:3.0/AV:N/AC:L/PR:N/UI:R/S:U/C:H/I:H/A:H","metrics":{"baseScore":8.8,"exploitabilityScore":2.9,"impactScore":5.9},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2024-6345","epss":0.00227,"percentile":0.45699,"date":"2025-06-03"}]}],"matchDetails":[{"type":"exact-direct-match","matcher":"python-matcher","searchedBy":{"language":"python","namespace":"github:language:python","package":{"name":"setuptools","version":"44.0.0"}},"found":{"versionConstraint":"<70.0.0 (python)","vulnerabilityID":"GHSA-cx63-2mw6-8hw5"},"fix":{"suggestedVersion":"70.0.0"}}],"artifact":{"id":"e32f7cbfb9974e4a","name":"setuptools","version":"44.0.0","type":"python","locations":[{"path":"/lib/python3.8/site-packages/setuptools-44.0.0.dist-info/METADATA","accessPath":"/lib/python3.8/site-packages/setuptools-44.0.0.dist-info/METADATA","annotations":{"evidence":"primary"}},{"path":"/lib/python3.8/site-packages/setuptools-44.0.0.dist-info/RECORD","accessPath":"/lib/python3.8/site-packages/setuptools-44.0.0.dist-info/RECORD","annotations":{"evidence":"supporting"}},{"path":"/lib/python3.8/site-packages/setuptools-44.0.0.dist-info/top_level.txt","accessPath":"/lib/python3.8/site-packages/setuptools-44.0.0.dist-info/top_level.txt","annotations":{"evidence":"supporting"}}],"language":"python","licenses":["UNKNOWN"],"cpes":["cpe:2.3:a:python:setuptools:44.0.0:*:*:*:*:*:*:*"],"purl":"pkg:pypi/setuptools@44.0.0","upstreams":[]}},{"vulnerability":{"id":"GHSA-5xp3-jfq3-5q8x","dataSource":"https://github.com/advisories/GHSA-5xp3-jfq3-5q8x","namespace":"github:language:python","severity":"High","urls":[],"description":"Improper Input Validation in pip","cvss":[{"type":"Secondary","version":"3.1","vector":"CVSS:3.1/AV:N/AC:L/PR:L/UI:R/S:U/C:N/I:H/A:N","metrics":{"baseScore":5.7,"exploitabilityScore":2.1,"impactScore":3.6},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2021-3572","epss":0.0024,"percentile":0.4753,"date":"2025-06-03"}],"fix":{"versions":["21.1"],"state":"fixed"},"advisories":[],"risk":0.15839999999999999},"relatedVulnerabilities":[{"id":"CVE-2021-3572","dataSource":"https://nvd.nist.gov/vuln/detail/CVE-2021-3572","namespace":"nvd:cpe","severity":"Medium","urls":["https://bugzilla.redhat.com/show_bug.cgi?id=1962856","https://security.netapp.com/advisory/ntap-20240621-0006/","https://www.oracle.com/security-alerts/cpuapr2022.html","https://www.oracle.com/security-alerts/cpujul2022.html","https://bugzilla.redhat.com/show_bug.cgi?id=1962856","https://security.netapp.com/advisory/ntap-20240621-0006/","https://www.oracle.com/security-alerts/cpuapr2022.html","https://www.oracle.com/security-alerts/cpujul2022.html"],"description":"A flaw was found in python-pip in the way it handled Unicode separators in git references. A remote attacker could possibly use this issue to install a different revision on a repository. The highest threat from this vulnerability is to data integrity. This is fixed in python-pip version 21.1.","cvss":[{"source":"nvd@nist.gov","type":"Primary","version":"3.1","vector":"CVSS:3.1/AV:N/AC:L/PR:L/UI:R/S:U/C:N/I:H/A:N","metrics":{"baseScore":5.7,"exploitabilityScore":2.1,"impactScore":3.6},"vendorMetadata":{}},{"source":"nvd@nist.gov","type":"Primary","version":"2.0","vector":"AV:N/AC:M/Au:S/C:N/I:P/A:N","metrics":{"baseScore":3.5,"exploitabilityScore":6.9,"impactScore":2.9},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2021-3572","epss":0.0024,"percentile":0.4753,"date":"2025-06-03"}]}],"matchDetails":[{"type":"exact-direct-match","matcher":"python-matcher","searchedBy":{"language":"python","namespace":"github:language:python","package":{"name":"pip","version":"20.0.2"}},"found":{"versionConstraint":"<21.1 (python)","vulnerabilityID":"GHSA-5xp3-jfq3-5q8x"},"fix":{"suggestedVersion":"21.1"}}],"artifact":{"id":"b110088862afdc1b","name":"pip","version":"20.0.2","type":"python","locations":[{"path":"/lib/python3.8/site-packages/pip-20.0.2.dist-info/METADATA","accessPath":"/lib/python3.8/site-packages/pip-20.0.2.dist-info/METADATA","annotations":{"evidence":"primary"}},{"path":"/lib/python3.8/site-packages/pip-20.0.2.dist-info/RECORD","accessPath":"/lib/python3.8/site-packages/pip-20.0.2.dist-info/RECORD","annotations":{"evidence":"supporting"}},{"path":"/lib/python3.8/site-packages/pip-20.0.2.dist-info/top_level.txt","accessPath":"/lib/python3.8/site-packages/pip-20.0.2.dist-info/top_level.txt","annotations":{"evidence":"supporting"}}],"language":"python","licenses":["MIT"],"cpes":["cpe:2.3:a:pip_developers_project:python-pip:20.0.2:*:*:*:*:*:*:*","cpe:2.3:a:pip_developers_project:python_pip:20.0.2:*:*:*:*:*:*:*","cpe:2.3:a:pip_developersproject:python-pip:20.0.2:*:*:*:*:*:*:*","cpe:2.3:a:pip_developersproject:python_pip:20.0.2:*:*:*:*:*:*:*","cpe:2.3:a:pypa_dev_project:python-pip:20.0.2:*:*:*:*:*:*:*","cpe:2.3:a:pypa_dev_project:python_pip:20.0.2:*:*:*:*:*:*:*","cpe:2.3:a:pip_developers_project:pip:20.0.2:*:*:*:*:*:*:*","cpe:2.3:a:pypa_devproject:python-pip:20.0.2:*:*:*:*:*:*:*","cpe:2.3:a:pypa_devproject:python_pip:20.0.2:*:*:*:*:*:*:*","cpe:2.3:a:pip_developers:python-pip:20.0.2:*:*:*:*:*:*:*","cpe:2.3:a:pip_developers:python_pip:20.0.2:*:*:*:*:*:*:*","cpe:2.3:a:pip_developersproject:pip:20.0.2:*:*:*:*:*:*:*","cpe:2.3:a:python-pip:python-pip:20.0.2:*:*:*:*:*:*:*","cpe:2.3:a:python-pip:python_pip:20.0.2:*:*:*:*:*:*:*","cpe:2.3:a:python_pip:python-pip:20.0.2:*:*:*:*:*:*:*","cpe:2.3:a:python_pip:python_pip:20.0.2:*:*:*:*:*:*:*","cpe:2.3:a:pypa_dev_project:pip:20.0.2:*:*:*:*:*:*:*","cpe:2.3:a:pypa-dev:python-pip:20.0.2:*:*:*:*:*:*:*","cpe:2.3:a:pypa-dev:python_pip:20.0.2:*:*:*:*:*:*:*","cpe:2.3:a:pypa_dev:python-pip:20.0.2:*:*:*:*:*:*:*","cpe:2.3:a:pypa_dev:python_pip:20.0.2:*:*:*:*:*:*:*","cpe:2.3:a:pypa_devproject:pip:20.0.2:*:*:*:*:*:*:*","cpe:2.3:a:pip_developers:pip:20.0.2:*:*:*:*:*:*:*","cpe:2.3:a:python:python-pip:20.0.2:*:*:*:*:*:*:*","cpe:2.3:a:python:python_pip:20.0.2:*:*:*:*:*:*:*","cpe:2.3:a:pypa:python-pip:20.0.2:*:*:*:*:*:*:*","cpe:2.3:a:pypa:python_pip:20.0.2:*:*:*:*:*:*:*","cpe:2.3:a:pip:python-pip:20.0.2:*:*:*:*:*:*:*","cpe:2.3:a:pip:python_pip:20.0.2:*:*:*:*:*:*:*","cpe:2.3:a:python-pip:pip:20.0.2:*:*:*:*:*:*:*","cpe:2.3:a:python_pip:pip:20.0.2:*:*:*:*:*:*:*","cpe:2.3:a:pypa-dev:pip:20.0.2:*:*:*:*:*:*:*","cpe:2.3:a:pypa_dev:pip:20.0.2:*:*:*:*:*:*:*","cpe:2.3:a:python:pip:20.0.2:*:*:*:*:*:*:*","cpe:2.3:a:pypa:pip:20.0.2:*:*:*:*:*:*:*","cpe:2.3:a:pip:pip:20.0.2:*:*:*:*:*:*:*"],"purl":"pkg:pypi/pip@20.0.2","upstreams":[]}},{"vulnerability":{"id":"GHSA-56wv-2wr9-3h9r","dataSource":"https://github.com/advisories/GHSA-56wv-2wr9-3h9r","namespace":"github:language:python","severity":"High","urls":[],"description":"Improper Verification of Cryptographic Signature in fastecdsa","cvss":[{"type":"Secondary","version":"3.1","vector":"CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:H/I:N/A:N","metrics":{"baseScore":7.5,"exploitabilityScore":3.9,"impactScore":3.6},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2020-12607","epss":0.00169,"percentile":0.39057,"date":"2025-06-03"}],"fix":{"versions":["2.1.2"],"state":"fixed"},"advisories":[],"risk":0.12675},"relatedVulnerabilities":[{"id":"CVE-2020-12607","dataSource":"https://nvd.nist.gov/vuln/detail/CVE-2020-12607","namespace":"nvd:cpe","severity":"High","urls":["https://github.com/AntonKueltz/fastecdsa/commit/4a16daeaf139be20654ef58a9fe4c79dc030458c","https://github.com/AntonKueltz/fastecdsa/commit/7b64e3efaa806b4daaf73bb5172af3581812f8de","https://github.com/AntonKueltz/fastecdsa/commit/e592f106edd5acf6dacedfab2ad16fe6c735c9d1","https://github.com/AntonKueltz/fastecdsa/issues/52","https://github.com/AntonKueltz/fastecdsa/commit/4a16daeaf139be20654ef58a9fe4c79dc030458c","https://github.com/AntonKueltz/fastecdsa/commit/7b64e3efaa806b4daaf73bb5172af3581812f8de","https://github.com/AntonKueltz/fastecdsa/commit/e592f106edd5acf6dacedfab2ad16fe6c735c9d1","https://github.com/AntonKueltz/fastecdsa/issues/52"],"description":"An issue was discovered in fastecdsa before 2.1.2. When using the NIST P-256 curve in the ECDSA implementation, the point at infinity is mishandled. This means that for an extreme value in k and s^-1, the signature verification fails even if the signature is correct. This behavior is not solely a usability problem. There are some threat models where an attacker can benefit by successfully guessing users for whom signature verification will fail.","cvss":[{"source":"nvd@nist.gov","type":"Primary","version":"3.1","vector":"CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:H/I:N/A:N","metrics":{"baseScore":7.5,"exploitabilityScore":3.9,"impactScore":3.6},"vendorMetadata":{}},{"source":"nvd@nist.gov","type":"Primary","version":"2.0","vector":"AV:N/AC:L/Au:N/C:P/I:N/A:N","metrics":{"baseScore":5,"exploitabilityScore":10,"impactScore":2.9},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2020-12607","epss":0.00169,"percentile":0.39057,"date":"2025-06-03"}]}],"matchDetails":[{"type":"exact-direct-match","matcher":"python-matcher","searchedBy":{"language":"python","namespace":"github:language:python","package":{"name":"fastecdsa","version":"1.7.4"}},"found":{"versionConstraint":"<2.1.2 (python)","vulnerabilityID":"GHSA-56wv-2wr9-3h9r"},"fix":{"suggestedVersion":"2.1.2"}}],"artifact":{"id":"5e9466701f95a8ac","name":"fastecdsa","version":"1.7.4","type":"python","locations":[{"path":"/lib/python3.8/site-packages/fastecdsa-1.7.4.dist-info/METADATA","accessPath":"/lib/python3.8/site-packages/fastecdsa-1.7.4.dist-info/METADATA","annotations":{"evidence":"primary"}},{"path":"/lib/python3.8/site-packages/fastecdsa-1.7.4.dist-info/RECORD","accessPath":"/lib/python3.8/site-packages/fastecdsa-1.7.4.dist-info/RECORD","annotations":{"evidence":"supporting"}},{"path":"/lib/python3.8/site-packages/fastecdsa-1.7.4.dist-info/top_level.txt","accessPath":"/lib/python3.8/site-packages/fastecdsa-1.7.4.dist-info/top_level.txt","annotations":{"evidence":"supporting"}}],"language":"python","licenses":["CC0 1.0 Universal (CC0 1.0) Public Domain Dedication"],"cpes":["cpe:2.3:a:anton_kueltz_project:python-fastecdsa:1.7.4:*:*:*:*:*:*:*","cpe:2.3:a:anton_kueltz_project:python_fastecdsa:1.7.4:*:*:*:*:*:*:*","cpe:2.3:a:kueltz_anton_project:python-fastecdsa:1.7.4:*:*:*:*:*:*:*","cpe:2.3:a:kueltz_anton_project:python_fastecdsa:1.7.4:*:*:*:*:*:*:*","cpe:2.3:a:anton_kueltzproject:python-fastecdsa:1.7.4:*:*:*:*:*:*:*","cpe:2.3:a:anton_kueltzproject:python_fastecdsa:1.7.4:*:*:*:*:*:*:*","cpe:2.3:a:kueltz_antonproject:python-fastecdsa:1.7.4:*:*:*:*:*:*:*","cpe:2.3:a:kueltz_antonproject:python_fastecdsa:1.7.4:*:*:*:*:*:*:*","cpe:2.3:a:python-fastecdsa:python-fastecdsa:1.7.4:*:*:*:*:*:*:*","cpe:2.3:a:python-fastecdsa:python_fastecdsa:1.7.4:*:*:*:*:*:*:*","cpe:2.3:a:python_fastecdsa:python-fastecdsa:1.7.4:*:*:*:*:*:*:*","cpe:2.3:a:python_fastecdsa:python_fastecdsa:1.7.4:*:*:*:*:*:*:*","cpe:2.3:a:anton_kueltz_project:fastecdsa:1.7.4:*:*:*:*:*:*:*","cpe:2.3:a:kueltz_anton_project:fastecdsa:1.7.4:*:*:*:*:*:*:*","cpe:2.3:a:anton_kueltz:python-fastecdsa:1.7.4:*:*:*:*:*:*:*","cpe:2.3:a:anton_kueltz:python_fastecdsa:1.7.4:*:*:*:*:*:*:*","cpe:2.3:a:anton_kueltzproject:fastecdsa:1.7.4:*:*:*:*:*:*:*","cpe:2.3:a:kueltz-anton:python-fastecdsa:1.7.4:*:*:*:*:*:*:*","cpe:2.3:a:kueltz-anton:python_fastecdsa:1.7.4:*:*:*:*:*:*:*","cpe:2.3:a:kueltz_anton:python-fastecdsa:1.7.4:*:*:*:*:*:*:*","cpe:2.3:a:kueltz_anton:python_fastecdsa:1.7.4:*:*:*:*:*:*:*","cpe:2.3:a:kueltz_antonproject:fastecdsa:1.7.4:*:*:*:*:*:*:*","cpe:2.3:a:fastecdsa:python-fastecdsa:1.7.4:*:*:*:*:*:*:*","cpe:2.3:a:fastecdsa:python_fastecdsa:1.7.4:*:*:*:*:*:*:*","cpe:2.3:a:python-fastecdsa:fastecdsa:1.7.4:*:*:*:*:*:*:*","cpe:2.3:a:python_fastecdsa:fastecdsa:1.7.4:*:*:*:*:*:*:*","cpe:2.3:a:python:python-fastecdsa:1.7.4:*:*:*:*:*:*:*","cpe:2.3:a:python:python_fastecdsa:1.7.4:*:*:*:*:*:*:*","cpe:2.3:a:anton_kueltz:fastecdsa:1.7.4:*:*:*:*:*:*:*","cpe:2.3:a:kueltz-anton:fastecdsa:1.7.4:*:*:*:*:*:*:*","cpe:2.3:a:kueltz_anton:fastecdsa:1.7.4:*:*:*:*:*:*:*","cpe:2.3:a:fastecdsa:fastecdsa:1.7.4:*:*:*:*:*:*:*","cpe:2.3:a:python:fastecdsa:1.7.4:*:*:*:*:*:*:*"],"purl":"pkg:pypi/fastecdsa@1.7.4","upstreams":[]}},{"vulnerability":{"id":"GHSA-ph86-g9r3-5qw4","dataSource":"https://github.com/advisories/GHSA-ph86-g9r3-5qw4","namespace":"github:language:python","severity":"High","urls":[],"description":"Uninitialized Variable in fastecdsa","cvss":[{"type":"Secondary","version":"3.1","vector":"CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:N/I:N/A:H","metrics":{"baseScore":7.5,"exploitabilityScore":3.9,"impactScore":3.6},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2024-21502","epss":0.00143,"percentile":0.35736,"date":"2025-06-03"}],"fix":{"versions":["2.3.2"],"state":"fixed"},"advisories":[],"risk":0.10725000000000001},"relatedVulnerabilities":[{"id":"CVE-2024-21502","dataSource":"https://nvd.nist.gov/vuln/detail/CVE-2024-21502","namespace":"nvd:cpe","severity":"High","urls":["https://gist.github.com/keltecc/49da037072276f21b005a8337c15db26","https://github.com/AntonKueltz/fastecdsa/blob/v2.3.1/src/curveMath.c%23L210","https://github.com/AntonKueltz/fastecdsa/commit/57fc5689c95d649dab7ef60cc99ac64589f01e36","https://security.snyk.io/vuln/SNYK-PYTHON-FASTECDSA-6262045","https://gist.github.com/keltecc/49da037072276f21b005a8337c15db26","https://github.com/AntonKueltz/fastecdsa/blob/v2.3.1/src/curveMath.c%23L210","https://github.com/AntonKueltz/fastecdsa/commit/57fc5689c95d649dab7ef60cc99ac64589f01e36","https://security.snyk.io/vuln/SNYK-PYTHON-FASTECDSA-6262045"],"description":"Versions of the package fastecdsa before 2.3.2 are vulnerable to Use of Uninitialized Variable on the stack, via the curvemath_mul function in src/curveMath.c, due to being used and interpreted as user-defined type. Depending on the variable's actual value it could be arbitrary free(), arbitrary realloc(), null pointer dereference and other. Since the stack can be controlled by the attacker, the vulnerability could be used to corrupt allocator structure, leading to possible heap exploitation. The attacker could cause denial of service by exploiting this vulnerability.","cvss":[{"source":"nvd@nist.gov","type":"Primary","version":"3.1","vector":"CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:N/I:N/A:H","metrics":{"baseScore":7.5,"exploitabilityScore":3.9,"impactScore":3.6},"vendorMetadata":{}},{"source":"report@snyk.io","type":"Secondary","version":"3.1","vector":"CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:N/I:N/A:H","metrics":{"baseScore":7.5,"exploitabilityScore":3.9,"impactScore":3.6},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2024-21502","epss":0.00143,"percentile":0.35736,"date":"2025-06-03"}]}],"matchDetails":[{"type":"exact-direct-match","matcher":"python-matcher","searchedBy":{"language":"python","namespace":"github:language:python","package":{"name":"fastecdsa","version":"1.7.4"}},"found":{"versionConstraint":"<2.3.2 (python)","vulnerabilityID":"GHSA-ph86-g9r3-5qw4"},"fix":{"suggestedVersion":"2.3.2"}}],"artifact":{"id":"5e9466701f95a8ac","name":"fastecdsa","version":"1.7.4","type":"python","locations":[{"path":"/lib/python3.8/site-packages/fastecdsa-1.7.4.dist-info/METADATA","accessPath":"/lib/python3.8/site-packages/fastecdsa-1.7.4.dist-info/METADATA","annotations":{"evidence":"primary"}},{"path":"/lib/python3.8/site-packages/fastecdsa-1.7.4.dist-info/RECORD","accessPath":"/lib/python3.8/site-packages/fastecdsa-1.7.4.dist-info/RECORD","annotations":{"evidence":"supporting"}},{"path":"/lib/python3.8/site-packages/fastecdsa-1.7.4.dist-info/top_level.txt","accessPath":"/lib/python3.8/site-packages/fastecdsa-1.7.4.dist-info/top_level.txt","annotations":{"evidence":"supporting"}}],"language":"python","licenses":["CC0 1.0 Universal (CC0 1.0) Public Domain Dedication"],"cpes":["cpe:2.3:a:anton_kueltz_project:python-fastecdsa:1.7.4:*:*:*:*:*:*:*","cpe:2.3:a:anton_kueltz_project:python_fastecdsa:1.7.4:*:*:*:*:*:*:*","cpe:2.3:a:kueltz_anton_project:python-fastecdsa:1.7.4:*:*:*:*:*:*:*","cpe:2.3:a:kueltz_anton_project:python_fastecdsa:1.7.4:*:*:*:*:*:*:*","cpe:2.3:a:anton_kueltzproject:python-fastecdsa:1.7.4:*:*:*:*:*:*:*","cpe:2.3:a:anton_kueltzproject:python_fastecdsa:1.7.4:*:*:*:*:*:*:*","cpe:2.3:a:kueltz_antonproject:python-fastecdsa:1.7.4:*:*:*:*:*:*:*","cpe:2.3:a:kueltz_antonproject:python_fastecdsa:1.7.4:*:*:*:*:*:*:*","cpe:2.3:a:python-fastecdsa:python-fastecdsa:1.7.4:*:*:*:*:*:*:*","cpe:2.3:a:python-fastecdsa:python_fastecdsa:1.7.4:*:*:*:*:*:*:*","cpe:2.3:a:python_fastecdsa:python-fastecdsa:1.7.4:*:*:*:*:*:*:*","cpe:2.3:a:python_fastecdsa:python_fastecdsa:1.7.4:*:*:*:*:*:*:*","cpe:2.3:a:anton_kueltz_project:fastecdsa:1.7.4:*:*:*:*:*:*:*","cpe:2.3:a:kueltz_anton_project:fastecdsa:1.7.4:*:*:*:*:*:*:*","cpe:2.3:a:anton_kueltz:python-fastecdsa:1.7.4:*:*:*:*:*:*:*","cpe:2.3:a:anton_kueltz:python_fastecdsa:1.7.4:*:*:*:*:*:*:*","cpe:2.3:a:anton_kueltzproject:fastecdsa:1.7.4:*:*:*:*:*:*:*","cpe:2.3:a:kueltz-anton:python-fastecdsa:1.7.4:*:*:*:*:*:*:*","cpe:2.3:a:kueltz-anton:python_fastecdsa:1.7.4:*:*:*:*:*:*:*","cpe:2.3:a:kueltz_anton:python-fastecdsa:1.7.4:*:*:*:*:*:*:*","cpe:2.3:a:kueltz_anton:python_fastecdsa:1.7.4:*:*:*:*:*:*:*","cpe:2.3:a:kueltz_antonproject:fastecdsa:1.7.4:*:*:*:*:*:*:*","cpe:2.3:a:fastecdsa:python-fastecdsa:1.7.4:*:*:*:*:*:*:*","cpe:2.3:a:fastecdsa:python_fastecdsa:1.7.4:*:*:*:*:*:*:*","cpe:2.3:a:python-fastecdsa:fastecdsa:1.7.4:*:*:*:*:*:*:*","cpe:2.3:a:python_fastecdsa:fastecdsa:1.7.4:*:*:*:*:*:*:*","cpe:2.3:a:python:python-fastecdsa:1.7.4:*:*:*:*:*:*:*","cpe:2.3:a:python:python_fastecdsa:1.7.4:*:*:*:*:*:*:*","cpe:2.3:a:anton_kueltz:fastecdsa:1.7.4:*:*:*:*:*:*:*","cpe:2.3:a:kueltz-anton:fastecdsa:1.7.4:*:*:*:*:*:*:*","cpe:2.3:a:kueltz_anton:fastecdsa:1.7.4:*:*:*:*:*:*:*","cpe:2.3:a:fastecdsa:fastecdsa:1.7.4:*:*:*:*:*:*:*","cpe:2.3:a:python:fastecdsa:1.7.4:*:*:*:*:*:*:*"],"purl":"pkg:pypi/fastecdsa@1.7.4","upstreams":[]}},{"vulnerability":{"id":"GHSA-qwmp-2cf2-g9g6","dataSource":"https://github.com/advisories/GHSA-qwmp-2cf2-g9g6","namespace":"github:language:python","severity":"High","urls":[],"description":"pypa/wheel vulnerable to Regular Expression denial of service (ReDoS)","cvss":[{"type":"Secondary","version":"3.1","vector":"CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:N/I:N/A:H","metrics":{"baseScore":7.5,"exploitabilityScore":3.9,"impactScore":3.6},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2022-40898","epss":0.00108,"percentile":0.30278,"date":"2025-06-03"}],"fix":{"versions":["0.38.1"],"state":"fixed"},"advisories":[],"risk":0.08099999999999999},"relatedVulnerabilities":[{"id":"CVE-2022-40898","dataSource":"https://nvd.nist.gov/vuln/detail/CVE-2022-40898","namespace":"nvd:cpe","severity":"High","urls":["https://github.com/pypa/wheel/blob/main/src/wheel/wheelfile.py#L18","https://pypi.org/project/wheel/","https://pyup.io/posts/pyup-discovers-redos-vulnerabilities-in-top-python-packages/","https://github.com/pypa/wheel/blob/main/src/wheel/wheelfile.py#L18","https://pypi.org/project/wheel/","https://pyup.io/posts/pyup-discovers-redos-vulnerabilities-in-top-python-packages/"],"description":"An issue discovered in Python Packaging Authority (PyPA) Wheel 0.37.1 and earlier allows remote attackers to cause a denial of service via attacker controlled input to wheel cli.","cvss":[{"source":"nvd@nist.gov","type":"Primary","version":"3.1","vector":"CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:N/I:N/A:H","metrics":{"baseScore":7.5,"exploitabilityScore":3.9,"impactScore":3.6},"vendorMetadata":{}},{"source":"134c704f-9b21-4f2e-91b3-4a467353bcc0","type":"Secondary","version":"3.1","vector":"CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:N/I:N/A:H","metrics":{"baseScore":7.5,"exploitabilityScore":3.9,"impactScore":3.6},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2022-40898","epss":0.00108,"percentile":0.30278,"date":"2025-06-03"}]}],"matchDetails":[{"type":"exact-direct-match","matcher":"python-matcher","searchedBy":{"language":"python","namespace":"github:language:python","package":{"name":"wheel","version":"0.34.2"}},"found":{"versionConstraint":"<0.38.1 (python)","vulnerabilityID":"GHSA-qwmp-2cf2-g9g6"},"fix":{"suggestedVersion":"0.38.1"}}],"artifact":{"id":"afe3ff06b290e130","name":"wheel","version":"0.34.2","type":"python","locations":[{"path":"/lib/python3.8/site-packages/wheel-0.34.2.dist-info/METADATA","accessPath":"/lib/python3.8/site-packages/wheel-0.34.2.dist-info/METADATA","annotations":{"evidence":"primary"}},{"path":"/lib/python3.8/site-packages/wheel-0.34.2.dist-info/RECORD","accessPath":"/lib/python3.8/site-packages/wheel-0.34.2.dist-info/RECORD","annotations":{"evidence":"supporting"}},{"path":"/lib/python3.8/site-packages/wheel-0.34.2.dist-info/top_level.txt","accessPath":"/lib/python3.8/site-packages/wheel-0.34.2.dist-info/top_level.txt","annotations":{"evidence":"supporting"}}],"language":"python","licenses":["MIT"],"cpes":["cpe:2.3:a:daniel_holth_project:python-wheel:0.34.2:*:*:*:*:*:*:*","cpe:2.3:a:daniel_holth_project:python_wheel:0.34.2:*:*:*:*:*:*:*","cpe:2.3:a:daniel_holthproject:python-wheel:0.34.2:*:*:*:*:*:*:*","cpe:2.3:a:daniel_holthproject:python_wheel:0.34.2:*:*:*:*:*:*:*","cpe:2.3:a:dholth_project:python-wheel:0.34.2:*:*:*:*:*:*:*","cpe:2.3:a:dholth_project:python_wheel:0.34.2:*:*:*:*:*:*:*","cpe:2.3:a:daniel_holth_project:wheel:0.34.2:*:*:*:*:*:*:*","cpe:2.3:a:dholthproject:python-wheel:0.34.2:*:*:*:*:*:*:*","cpe:2.3:a:dholthproject:python_wheel:0.34.2:*:*:*:*:*:*:*","cpe:2.3:a:daniel_holth:python-wheel:0.34.2:*:*:*:*:*:*:*","cpe:2.3:a:daniel_holth:python_wheel:0.34.2:*:*:*:*:*:*:*","cpe:2.3:a:daniel_holthproject:wheel:0.34.2:*:*:*:*:*:*:*","cpe:2.3:a:python-wheel:python-wheel:0.34.2:*:*:*:*:*:*:*","cpe:2.3:a:python-wheel:python_wheel:0.34.2:*:*:*:*:*:*:*","cpe:2.3:a:python_wheel:python-wheel:0.34.2:*:*:*:*:*:*:*","cpe:2.3:a:python_wheel:python_wheel:0.34.2:*:*:*:*:*:*:*","cpe:2.3:a:dholth_project:wheel:0.34.2:*:*:*:*:*:*:*","cpe:2.3:a:dholth:python-wheel:0.34.2:*:*:*:*:*:*:*","cpe:2.3:a:dholth:python_wheel:0.34.2:*:*:*:*:*:*:*","cpe:2.3:a:dholthproject:wheel:0.34.2:*:*:*:*:*:*:*","cpe:2.3:a:python:python-wheel:0.34.2:*:*:*:*:*:*:*","cpe:2.3:a:python:python_wheel:0.34.2:*:*:*:*:*:*:*","cpe:2.3:a:daniel_holth:wheel:0.34.2:*:*:*:*:*:*:*","cpe:2.3:a:python-wheel:wheel:0.34.2:*:*:*:*:*:*:*","cpe:2.3:a:python_wheel:wheel:0.34.2:*:*:*:*:*:*:*","cpe:2.3:a:wheel:python-wheel:0.34.2:*:*:*:*:*:*:*","cpe:2.3:a:wheel:python_wheel:0.34.2:*:*:*:*:*:*:*","cpe:2.3:a:dholth:wheel:0.34.2:*:*:*:*:*:*:*","cpe:2.3:a:python:wheel:0.34.2:*:*:*:*:*:*:*","cpe:2.3:a:wheel:wheel:0.34.2:*:*:*:*:*:*:*"],"purl":"pkg:pypi/wheel@0.34.2","upstreams":[]}},{"vulnerability":{"id":"GHSA-mq26-g339-26xf","dataSource":"https://github.com/advisories/GHSA-mq26-g339-26xf","namespace":"github:language:python","severity":"Medium","urls":[],"description":"Command Injection in pip when used with Mercurial","cvss":[{"type":"Secondary","version":"3.1","vector":"CVSS:3.1/AV:L/AC:L/PR:L/UI:N/S:U/C:N/I:H/A:N","metrics":{"baseScore":5.5,"exploitabilityScore":1.9,"impactScore":3.6},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2023-5752","epss":0.0004,"percentile":0.11691,"date":"2025-06-03"}],"fix":{"versions":["23.3"],"state":"fixed"},"advisories":[],"risk":0.021},"relatedVulnerabilities":[{"id":"CVE-2023-5752","dataSource":"https://nvd.nist.gov/vuln/detail/CVE-2023-5752","namespace":"nvd:cpe","severity":"Low","urls":["https://github.com/pypa/pip/pull/12306","https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/622OZXWG72ISQPLM5Y57YCVIMWHD4C3U/","https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/65UKKF5LBHEFDCUSPBHUN4IHYX7SRMHH/","https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/FXUVMJM25PUAZRQZBF54OFVKTY3MINPW/","https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/KFC2SPFG5FLCZBYY2K3T5MFW2D22NG6E/","https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/YBSB3SUPQ3VIFYUMHPO3MEQI4BJAXKCZ/","https://mail.python.org/archives/list/security-announce@python.org/thread/F4PL35U6X4VVHZ5ILJU3PWUWN7H7LZXL/","https://github.com/pypa/pip/pull/12306","https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/622OZXWG72ISQPLM5Y57YCVIMWHD4C3U/","https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/65UKKF5LBHEFDCUSPBHUN4IHYX7SRMHH/","https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/FXUVMJM25PUAZRQZBF54OFVKTY3MINPW/","https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/KFC2SPFG5FLCZBYY2K3T5MFW2D22NG6E/","https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/YBSB3SUPQ3VIFYUMHPO3MEQI4BJAXKCZ/","https://mail.python.org/archives/list/security-announce@python.org/thread/F4PL35U6X4VVHZ5ILJU3PWUWN7H7LZXL/"],"description":"When installing a package from a Mercurial VCS URL  (ie \"pip install \nhg+...\") with pip prior to v23.3, the specified Mercurial revision could\n be used to inject arbitrary configuration options to the \"hg clone\" \ncall (ie \"--config\"). Controlling the Mercurial configuration can modify\n how and which repository is installed. This vulnerability does not \naffect users who aren't installing from Mercurial.","cvss":[{"source":"nvd@nist.gov","type":"Primary","version":"3.1","vector":"CVSS:3.1/AV:L/AC:L/PR:L/UI:N/S:U/C:N/I:L/A:N","metrics":{"baseScore":3.3,"exploitabilityScore":1.9,"impactScore":1.5},"vendorMetadata":{}},{"source":"cna@python.org","type":"Secondary","version":"3.1","vector":"CVSS:3.1/AV:L/AC:L/PR:L/UI:N/S:U/C:N/I:H/A:N","metrics":{"baseScore":5.5,"exploitabilityScore":1.9,"impactScore":3.6},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2023-5752","epss":0.0004,"percentile":0.11691,"date":"2025-06-03"}]}],"matchDetails":[{"type":"exact-direct-match","matcher":"python-matcher","searchedBy":{"language":"python","namespace":"github:language:python","package":{"name":"pip","version":"20.0.2"}},"found":{"versionConstraint":"<23.3 (python)","vulnerabilityID":"GHSA-mq26-g339-26xf"},"fix":{"suggestedVersion":"23.3"}}],"artifact":{"id":"b110088862afdc1b","name":"pip","version":"20.0.2","type":"python","locations":[{"path":"/lib/python3.8/site-packages/pip-20.0.2.dist-info/METADATA","accessPath":"/lib/python3.8/site-packages/pip-20.0.2.dist-info/METADATA","annotations":{"evidence":"primary"}},{"path":"/lib/python3.8/site-packages/pip-20.0.2.dist-info/RECORD","accessPath":"/lib/python3.8/site-packages/pip-20.0.2.dist-info/RECORD","annotations":{"evidence":"supporting"}},{"path":"/lib/python3.8/site-packages/pip-20.0.2.dist-info/top_level.txt","accessPath":"/lib/python3.8/site-packages/pip-20.0.2.dist-info/top_level.txt","annotations":{"evidence":"supporting"}}],"language":"python","licenses":["MIT"],"cpes":["cpe:2.3:a:pip_developers_project:python-pip:20.0.2:*:*:*:*:*:*:*","cpe:2.3:a:pip_developers_project:python_pip:20.0.2:*:*:*:*:*:*:*","cpe:2.3:a:pip_developersproject:python-pip:20.0.2:*:*:*:*:*:*:*","cpe:2.3:a:pip_developersproject:python_pip:20.0.2:*:*:*:*:*:*:*","cpe:2.3:a:pypa_dev_project:python-pip:20.0.2:*:*:*:*:*:*:*","cpe:2.3:a:pypa_dev_project:python_pip:20.0.2:*:*:*:*:*:*:*","cpe:2.3:a:pip_developers_project:pip:20.0.2:*:*:*:*:*:*:*","cpe:2.3:a:pypa_devproject:python-pip:20.0.2:*:*:*:*:*:*:*","cpe:2.3:a:pypa_devproject:python_pip:20.0.2:*:*:*:*:*:*:*","cpe:2.3:a:pip_developers:python-pip:20.0.2:*:*:*:*:*:*:*","cpe:2.3:a:pip_developers:python_pip:20.0.2:*:*:*:*:*:*:*","cpe:2.3:a:pip_developersproject:pip:20.0.2:*:*:*:*:*:*:*","cpe:2.3:a:python-pip:python-pip:20.0.2:*:*:*:*:*:*:*","cpe:2.3:a:python-pip:python_pip:20.0.2:*:*:*:*:*:*:*","cpe:2.3:a:python_pip:python-pip:20.0.2:*:*:*:*:*:*:*","cpe:2.3:a:python_pip:python_pip:20.0.2:*:*:*:*:*:*:*","cpe:2.3:a:pypa_dev_project:pip:20.0.2:*:*:*:*:*:*:*","cpe:2.3:a:pypa-dev:python-pip:20.0.2:*:*:*:*:*:*:*","cpe:2.3:a:pypa-dev:python_pip:20.0.2:*:*:*:*:*:*:*","cpe:2.3:a:pypa_dev:python-pip:20.0.2:*:*:*:*:*:*:*","cpe:2.3:a:pypa_dev:python_pip:20.0.2:*:*:*:*:*:*:*","cpe:2.3:a:pypa_devproject:pip:20.0.2:*:*:*:*:*:*:*","cpe:2.3:a:pip_developers:pip:20.0.2:*:*:*:*:*:*:*","cpe:2.3:a:python:python-pip:20.0.2:*:*:*:*:*:*:*","cpe:2.3:a:python:python_pip:20.0.2:*:*:*:*:*:*:*","cpe:2.3:a:pypa:python-pip:20.0.2:*:*:*:*:*:*:*","cpe:2.3:a:pypa:python_pip:20.0.2:*:*:*:*:*:*:*","cpe:2.3:a:pip:python-pip:20.0.2:*:*:*:*:*:*:*","cpe:2.3:a:pip:python_pip:20.0.2:*:*:*:*:*:*:*","cpe:2.3:a:python-pip:pip:20.0.2:*:*:*:*:*:*:*","cpe:2.3:a:python_pip:pip:20.0.2:*:*:*:*:*:*:*","cpe:2.3:a:pypa-dev:pip:20.0.2:*:*:*:*:*:*:*","cpe:2.3:a:pypa_dev:pip:20.0.2:*:*:*:*:*:*:*","cpe:2.3:a:python:pip:20.0.2:*:*:*:*:*:*:*","cpe:2.3:a:pypa:pip:20.0.2:*:*:*:*:*:*:*","cpe:2.3:a:pip:pip:20.0.2:*:*:*:*:*:*:*"],"purl":"pkg:pypi/pip@20.0.2","upstreams":[]}}],"source":{"type":"sbom-file","target":"validity-sensors-tools-syft.json"},"distro":{"name":"","version":"","idLike":null},"descriptor":{"name":"grype","version":"0.92.2","configuration":{"output":["json=/Users/alan/Personal/popey/syft-snap-source-test/grype/v1.26.0/2025-06-04/validity-sensors-tools-grype.json","template=/Users/alan/Personal/popey/syft-snap-source-test/grype/v1.26.0/2025-06-04/validity-sensors-tools-grype.html"],"file":"","pretty":false,"distro":"","add-cpes-if-none":false,"output-template-file":"../html.tmpl","check-for-app-update":true,"only-fixed":false,"only-notfixed":false,"ignore-wontfix":"","platform":"","search":{"scope":"squashed","unindexed-archives":false,"indexed-archives":true},"ignore":[{"vulnerability":"","reason":"","namespace":"","fix-state":"","package":{"name":"kernel-headers","version":"","language":"","type":"rpm","location":"","upstream-name":"kernel"},"vex-status":"","vex-justification":"","match-type":"exact-indirect-match"},{"vulnerability":"","reason":"","namespace":"","fix-state":"","package":{"name":"linux(-.*)?-headers-.*","version":"","language":"","type":"deb","location":"","upstream-name":"linux.*"},"vex-status":"","vex-justification":"","match-type":"exact-indirect-match"},{"vulnerability":"","reason":"","namespace":"","fix-state":"","package":{"name":"linux-libc-dev","version":"","language":"","type":"deb","location":"","upstream-name":"linux"},"vex-status":"","vex-justification":"","match-type":"exact-indirect-match"}],"exclude":[],"externalSources":{"enable":false,"maven":{"searchUpstreamBySha1":true,"baseUrl":"https://search.maven.org/solrsearch/select","rateLimit":300000000}},"match":{"java":{"using-cpes":false},"jvm":{"using-cpes":true},"dotnet":{"using-cpes":false},"golang":{"using-cpes":false,"always-use-cpe-for-stdlib":true,"allow-main-module-pseudo-version-comparison":false},"javascript":{"using-cpes":false},"python":{"using-cpes":false},"ruby":{"using-cpes":false},"rust":{"using-cpes":false},"stock":{"using-cpes":true}},"fail-on-severity":"","registry":{"insecure-skip-tls-verify":false,"insecure-use-http":false,"auth":null,"ca-cert":""},"show-suppressed":false,"by-cve":false,"SortBy":{"sort-by":"risk"},"name":"","default-image-pull-source":"","vex-documents":[],"vex-add":[],"match-upstream-kernel-headers":false,"db":{"cache-dir":"/Users/alan/Library/Caches/grype/db","update-url":"https://grype.anchore.io/databases","ca-cert":"","auto-update":true,"validate-by-hash-on-start":true,"validate-age":true,"max-allowed-built-age":432000000000000,"require-update-check":false,"update-available-timeout":30000000000,"update-download-timeout":300000000000,"max-update-check-frequency":7200000000000},"exp":{},"dev":{"db":{"debug":false}}},"db":{"status":{"schemaVersion":"v6.0.2","from":"https://grype.anchore.io/databases/v6/vulnerability-db_v6.0.2_2025-06-04T01:32:23Z_1749035557.tar.zst?checksum=sha256%3A77238f4ffbc7cf72bf4dedabbb21a3d0dd3ade90fb5d9df7c4da2870f8d1a9b7","built":"2025-06-04T11:12:37Z","path":"/Users/alan/Library/Caches/grype/db/6/vulnerability.db","valid":true},"providers":{"alpine":{"captured":"2025-06-04T01:32:23Z","input":"xxh64:a6a065ece392f6d1"},"amazon":{"captured":"2025-06-04T01:32:32Z","input":"xxh64:b65d458d6858d72a"},"chainguard":{"captured":"2025-06-04T01:32:33Z","input":"xxh64:03c9c4d182d77791"},"debian":{"captured":"2025-06-04T01:32:49Z","input":"xxh64:169ceb8fbcb8d6f1"},"epss":{"captured":"2025-06-04T01:32:42Z","input":"xxh64:bf87c4c234d02346"},"github":{"captured":"2025-06-04T01:32:30Z","input":"xxh64:3dab0be50635e057"},"kev":{"captured":"2025-06-04T01:32:45Z","input":"xxh64:a934e75f468f4e09"},"mariner":{"captured":"2025-06-04T01:32:43Z","input":"xxh64:ac698752085c186f"},"nvd":{"captured":"2025-06-04T01:33:28Z","input":"xxh64:389887843a59d29c"},"oracle":{"captured":"2025-06-04T01:32:35Z","input":"xxh64:0b8aabcfb6505b62"},"rhel":{"captured":"2025-06-04T04:49:10Z","input":"xxh64:947d0caeb19e9ccc"},"sles":{"captured":"2025-06-04T01:32:34Z","input":"xxh64:1c50b1219296628a"},"ubuntu":{"captured":"2025-06-04T01:33:14Z","input":"xxh64:71e22d5c5bbe7d14"},"wolfi":{"captured":"2025-06-04T01:32:42Z","input":"xxh64:e68fca5a870c6769"}}},"timestamp":"2025-06-04T19:49:53.379748+01:00"}}
